src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -79,6 +79,12 @@ public override void Initialize (AnalysisContext context)
 					Location location)
 				{
 					var attributes = method.GetAttributes ();
+					string methodName;
+					if (method.MethodKind == MethodKind.PropertyGet || method.MethodKind == MethodKind.PropertySet)
```

Comment:
This is a really hot path -- this code is going to get run for basically every single method call in the program.

We should follow some guidelines:

1. Only do work if it's a method we're interested in (has the right attributes)
2. Avoid allocation as much as possible. That generally means no LINQ.

I'd move this entire block into the `foreach` loop, and let'se see if we can minimize the string concatenation, maybe by using a `StringBuilder`